build/edge/Dockerfile

```diff
@@ -27,4 +27,6 @@ VOLUME ["/etc/kubeedge/certs", "/var/lib/edged", "/var/lib/kubeedge", "/var/run/
 COPY --from=builder /usr/local/bin/edgecore /usr/local/bin/edgecore
 COPY --from=builder /go/src/github.com/kubeedge/kubeedge/edge/conf /etc/kubeedge/edge/conf
 
+EXPOSE 1884
```

Comment:
If we don't expose port here, the device can't access the 1884 in container?

Reply:
Only if we expose the port here, devices can access the mqtt server **in container and in the same container network**. If we want to access the mqtt server on the host, we still need to map the port on the host. Then [build/edge/docker-compose.yaml](https://github.com/kubeedge/kubeedge/blob/master/build/edge/docker-compose.yaml) or [build/edge/kubernetes/04-deployment-edgenode.yaml](https://github.com/kubeedge/kubeedge/blob/master/build/edge/kubernetes/04-deployment-edgenode.yaml) should be modified to map the ports.

Reply:
if devices and mqtt server are in the same container network ，device can directly access 127.0.0.1: 1884.

Reply:
> Only if we expose the port here, devices can access the mqtt server in container and in the same container network. If we want to access the mqtt server on the host,

do you mean device and mqttserver are in same pod ?

Reply:
> > Only if we expose the port here, devices can access the mqtt server in container and in the same container network. If we want to access the mqtt server on the host,
> 
> do you mean device and mqttserver are in same pod ?

Yes. Or if the edgecore is started by docker-compose as is shown in [build/edge/docker-compose.yaml](https://github.com/kubeedge/kubeedge/blob/master/build/edge/docker-compose.yaml), the device should be started by the network "kubeedge-bridge".

I know it looks like strange. If we want to deploy device separately, we should map the ports on the host.

Anyway, 1884 port should be exposed :)

Reply:
> do you mean device and mqttserver are in same pod ?

Why device and mqttserver could run in same pod？mqttserver run in edgecore, device is another process, if we don't expose the 1884 port, can't we access the mqttserver through containerIP:1884?

Reply:
> > do you mean device and mqttserver are in same pod ?
> 
> Why device and mqttserver could run in same pod？mqttserver run in edgecore, device is another process, if we don't expose the 1884 port, can't we access the mqttserver through containerIP:1884?

Sorry, my fault. I forgot the containerIP method. And I checked the offical docs of EXPOSE carefully, [EXPOSE](https://docs.docker.com/engine/reference/builder/#expose), then I found I misunderstood the meaning of EXPOSE. Sorry for wating your time, thanks!

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. To actually publish the port when running the container, use the -p flag on docker run to publish and map one or more ports, or the -P flag to publish all exposed ports and map them to high-order ports.

Reply:
> And I checked the offical docs of EXPOSE carefully, EXPOSE, then I found I misunderstood the meaning of EXPOSE. Sorry for wating your time, thanks!

Np @clumsy456 If you want to run the edgecore as k8s container, suggest to use the `hostNetwork` mode.  And I'd suggest you run edgecore as process instead of container, because it's related the runtime(docker)、storage in host. :)

Reply:
> > And I checked the offical docs of EXPOSE carefully, EXPOSE, then I found I misunderstood the meaning of EXPOSE. Sorry for wating your time, thanks!
> 
> Np @clumsy456 If you want to run the edgecore as k8s container, suggest to use the `hostNetwork` mode. And I'd suggest you run edgecore as process instead of container, because it's related the runtime(docker)、storage in host. :)

Thank you so much for your recommandation! Actually I tried deployment, docker-compose, and process.

Reply:
OK, you can delete the 1st commit, let's got 2nd merged, thanks!

Reply:
I've revert the first commit, and changed the title and release-node in the first comment.